File: exercise_b.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Inverse Theory - Exercise B
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%
% Loading the data and helper functions.
%%%

addpath('utils/matlab/');
load('data/matlab/exercise_b.mat');

%%%
%
% Plotting the distribution in a nice way is bit tricky
% because it spans several orders of magnitude but also
% 0.0. To circumvent this you can replace zero values with
% a very small non-zero value.
%
%%%

%%%
%
% Tropical Storm Saola
%
% The plot_modis and plot_gmi_swath functions are provided to plot the
% MODIS image and your IWP retrieval results, respectively.
%
% Your retrieval will contain a lot of very small values.
% To avoid them from messing up the color scale it's a good idea
% to restrict the range of values plotted to significant IWP
% values.
%
% You also need to pass the index of the GMI orbit
% to the plot functions according to whether
% you computed the IWP for gmi_tbs_0 (orbit_index = 0)
% or gmi_tbs_1 (orbit_index = 1).
%
%%%


subplot(2, 2, 1);
plot_modis();

% Mean
subplot(2, 2, 2);
iwp = max(iwp_mean, 1e-3);
title('Mean');
% Plotting IWP for gmi_tbs_0.
plot_gmi_swath(log(iwp), 0);
colorbar();

% Median
subplot(2, 2, 3);
title('Median');
iwp = max(iwp_median, 1e-3);
% Plotting IWP for gmi_tbs_0.
plot_gmi_swath(log(iwp), 0);
colorbar();

subplot(2, 2, 4);
title('10th Percentile');
iwp = max(iwp_10_p, 1e-3);
% Plotting IWP for gmi_tbs_0.
plot_gmi_swath(log(iwp), 0);
colorbar();

File: exercise_b.py
import numpy as np
import utils.setup
import matplotlib.pyplot as plt

##########################################################################################
# Inverse Theory - Exercise B
##########################################################################################

#
# Loading the data
#
y_database   = np.load("data/python/y_database.npy")
iwp_database = np.load("data/python/iwp_database.npy")


#
# Plotting the distribution in a nice way is bit tricky
# because it spans several orders of magnitude but also
# 0.0. To circumvent this you can either
#
# - Create log-spaced histogram bins yourself and set the
#   leftmost to zero.
# - Replace zero values with a very small non-zero value.


#
# Tropical Storm Saola
#

from utils.plots import plot_modis_image, plot_gmi_swath
from matplotlib.colors import LogNorm

# The plot_modis_image and plot_gmi_swath functions are
# provided to simplify plotting of your retrieval results.
#
# Your retrieval will contian a lot of very small values
# to avoid them from messing up the scale and also because
# the IWP values span several orders of magnitude, it's
# a good idea to use a custom log-norm for all plots.

norm = LogNorm(vmin = 1e-3, vmax = 100.0)
f, axs = plt.subplots(2, 2, figsize = (8, 8))


plot_modis_image(ax = axs[0, 0])
axs[0, 0].set_title("Modis RGB")

# Mean IWP
im = plot_gmi_swath(iwp_mean, orbit_index = 0, ax = axs[0, 1], norm = norm)
axs[0, 1].set_title("Mean")

# Median
plot_gmi_swath(iwp_median, orbit_index = 0, ax = axs[1, 0], norm = norm)
axs[1, 0].set_title("Median")

# Median
plot_gmi_swath(iwp_10_p, orbit_index = 0, ax = axs[1, 0], norm = norm)
axs[1, 0].set_title("10th Percentile")

plt.colorbar(im, ax = axs.ravel().tolist())
